docs: allow to define alerts for a consumer (#424)
gravitee-io/gravitee-docs: pages/ae/apim/apim-consumer.adoc

= API
:page-sidebar: ae_sidebar
:page-permalink: ae/apim_consumer.html
:page-folder: ae/apim
:page-description: Gravitee Alert Engine - API Management - Consumer
:page-toc: true
:page-keywords: Gravitee, API Platform, Alert, Alert Engine, documentation, manual, guide, reference, api
:page-layout: ae
:page-liquid:

== Overview

> This section explains how to configure alerts for a consumer through applications in APIM.
> 
> This configuration takes place in APIM Portal. You can create a maximum of 10 alerts by application.

Reviewer: I'm not sure of the formulation
```suggestion
This section explains how to configure alerts for a consumer through applications in APIM.

You can set up these alerts in APIM Portal and create a maximum of 10 alerts by application.
```

.API Management - Consumer Alerts
image::ae/apim/consumer_alerts.png[Consumer alerts]

== Notification

When creating an alert for an application, all the members of the applications (direct members or through associated groups) with a configured email will receive an email notification if an alert is triggered.

== Type of alerts

You can configure two types of alerts.

=== Response time greater than X ms

> This example configures an alert for response times for requests exceeding a threshold of 1500ms during the last 5 minutes:

Reviewer: ```suggestion
This example shows the configuration of an alert for response times exceeding a threshold of 1500 ms in the last 5 minutes:
```

.API Management - Response time > 1500ms
image::ae/apim/consumer_alert_response_time_threshold.png[Response time > 1500ms]

=== Rate of HTTP statuses during a period

You can configure alert on ranges of statuses (1XX, 2XX, 3XX, 4XX, 5XX) or directly on a precise status.

> This example configures an alert for a rate of HTTP status 5XX (Server error) greater than 75% during the last 10 minutes:

Reviewer: ```suggestion
This example shows the configuration an alert for a rate of HTTP status 5XX (Server error) greater than 75% during the last 10 minutes:
```
